File: pheno_pkg/src/java/PhenologyState.java
import  java.io.*;
import  java.util.*;
public class PhenologyState
{
    private double leafNumber;
    private double minFinalNumber;
    private List<Date> calendarDates;
    private List<String> calendarMoments;
    private double ptq;
    private double gAImean;
    private List<Double> listTTShootWindowForPTQ;
    private double pastMaxAI;
    private List<Double> listGAITTWindowForPTQ;
    private List<Double> listPARTTWindowForPTQ;
    private List<Double> calendarCumuls;
    private double vernaprog;
    private int hasLastPrimordiumAppeared;
    private double phase;
    private double finalLeafNumber;
    private int hasZadokStageChanged;
    private String currentZadokStage;
    private int hasFlagLeafLiguleAppeared;
    private List<Double> tilleringProfile;
    private List<Integer> leafTillerNumberArray;
    private int tillerNumber;
    private double canopyShootNumber;
    private double phyllochron;
    private double averageShootNumberPerPlant;
    private int isMomentRegistredZC_39;
    
    public PhenologyState()
    {
           
    }
    
    public PhenologyState(PhenologyState toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            this.leafNumber = toCopy.leafNumber;
            this.minFinalNumber = toCopy.minFinalNumber;
            List <Date> _calendarDates = new ArrayList<>();
            for (Date c : toCopy.calendarDates)
            {
                _calendarDates.add(c);
            }
            this.calendarDates = _calendarDates;
            List <String> _calendarMoments = new ArrayList<>();
            for (String c : toCopy.calendarMoments)
            {
                _calendarMoments.add(c);
            }
            this.calendarMoments = _calendarMoments;
            this.ptq = toCopy.ptq;
            this.gAImean = toCopy.gAImean;
            List <Double> _listTTShootWindowForPTQ = new ArrayList<>();
            for (Double c : toCopy.listTTShootWindowForPTQ)
            {
                _listTTShootWindowForPTQ.add(c);
            }
            this.listTTShootWindowForPTQ = _listTTShootWindowForPTQ;
            this.pastMaxAI = toCopy.pastMaxAI;
            List <Double> _listGAITTWindowForPTQ = new ArrayList<>();
            for (Double c : toCopy.listGAITTWindowForPTQ)
            {
                _listGAITTWindowForPTQ.add(c);
            }
            this.listGAITTWindowForPTQ = _listGAITTWindowForPTQ;
            List <Double> _listPARTTWindowForPTQ = new ArrayList<>();
            for (Double c : toCopy.listPARTTWindowForPTQ)
            {
                _listPARTTWindowForPTQ.add(c);
            }
            this.listPARTTWindowForPTQ = _listPARTTWindowForPTQ;
            List <Double> _calendarCumuls = new ArrayList<>();
            for (Double c : toCopy.calendarCumuls)
            {
                _calendarCumuls.add(c);
            }
            this.calendarCumuls = _calendarCumuls;
            this.vernaprog = toCopy.vernaprog;
            this.hasLastPrimordiumAppeared = toCopy.hasLastPrimordiumAppeared;
            this.phase = toCopy.phase;
            this.finalLeafNumber = toCopy.finalLeafNumber;
            this.hasZadokStageChanged = toCopy.hasZadokStageChanged;
            this.currentZadokStage = toCopy.currentZadokStage;
            this.hasFlagLeafLiguleAppeared = toCopy.hasFlagLeafLiguleAppeared;
            List <Double> _tilleringProfile = new ArrayList<>();
            for (Double c : toCopy.tilleringProfile)
            {
                _tilleringProfile.add(c);
            }
            this.tilleringProfile = _tilleringProfile;
            List <Integer> _leafTillerNumberArray = new ArrayList<>();
            for (Integer c : toCopy.leafTillerNumberArray)
            {
                _leafTillerNumberArray.add(c);
            }
            this.leafTillerNumberArray = _leafTillerNumberArray;
            this.tillerNumber = toCopy.tillerNumber;
            this.canopyShootNumber = toCopy.canopyShootNumber;
            this.phyllochron = toCopy.phyllochron;
            this.averageShootNumberPerPlant = toCopy.averageShootNumberPerPlant;
            this.isMomentRegistredZC_39 = toCopy.isMomentRegistredZC_39;
        }
    }
    public double getleafNumber()
    {
        return leafNumber;
    }

    public void setleafNumber(double _leafNumber)
    {
        this.leafNumber= _leafNumber;
    } 
    
    public double getminFinalNumber()
    {
        return minFinalNumber;
    }

    public void setminFinalNumber(double _minFinalNumber)
    {
        this.minFinalNumber= _minFinalNumber;
    } 
    
    public List<Date> getcalendarDates()
    {
        return calendarDates;
    }

    public void setcalendarDates(List<Date> _calendarDates)
    {
        this.calendarDates= _calendarDates;
    } 
    
    public List<String> getcalendarMoments()
    {
        return calendarMoments;
    }

    public void setcalendarMoments(List<String> _calendarMoments)
    {
        this.calendarMoments= _calendarMoments;
    } 
    
    public double getptq()
    {
        return ptq;
    }

    public void setptq(double _ptq)
    {
        this.ptq= _ptq;
    } 
    
    public double getgAImean()
    {
        return gAImean;
    }

    public void setgAImean(double _gAImean)
    {
        this.gAImean= _gAImean;
    } 
    
    public List<Double> getlistTTShootWindowForPTQ()
    {
        return listTTShootWindowForPTQ;
    }

    public void setlistTTShootWindowForPTQ(List<Double> _listTTShootWindowForPTQ)
    {
        this.listTTShootWindowForPTQ= _listTTShootWindowForPTQ;
    } 
    
    public double getpastMaxAI()
    {
        return pastMaxAI;
    }

    public void setpastMaxAI(double _pastMaxAI)
    {
        this.pastMaxAI= _pastMaxAI;
    } 
    
    public List<Double> getlistGAITTWindowForPTQ()
    {
        return listGAITTWindowForPTQ;
    }

    public void setlistGAITTWindowForPTQ(List<Double> _listGAITTWindowForPTQ)
    {
        this.listGAITTWindowForPTQ= _listGAITTWindowForPTQ;
    } 
    
    public List<Double> getlistPARTTWindowForPTQ()
    {
        return listPARTTWindowForPTQ;
    }

    public void setlistPARTTWindowForPTQ(List<Double> _listPARTTWindowForPTQ)
    {
        this.listPARTTWindowForPTQ= _listPARTTWindowForPTQ;
    } 
    
    public List<Double> getcalendarCumuls()
    {
        return calendarCumuls;
    }

    public void setcalendarCumuls(List<Double> _calendarCumuls)
    {
        this.calendarCumuls= _calendarCumuls;
    } 
    
    public double getvernaprog()
    {
        return vernaprog;
    }

    public void setvernaprog(double _vernaprog)
    {
        this.vernaprog= _vernaprog;
    } 
    
    public int gethasLastPrimordiumAppeared()
    {
        return hasLastPrimordiumAppeared;
    }

    public void sethasLastPrimordiumAppeared(int _hasLastPrimordiumAppeared)
    {
        this.hasLastPrimordiumAppeared= _hasLastPrimordiumAppeared;
    } 
    
    public double getphase()
    {
        return phase;
    }

    public void setphase(double _phase)
    {
        this.phase= _phase;
    } 
    
    public double getfinalLeafNumber()
    {
        return finalLeafNumber;
    }

    public void setfinalLeafNumber(double _finalLeafNumber)
    {
        this.finalLeafNumber= _finalLeafNumber;
    } 
    
    public int gethasZadokStageChanged()
    {
        return hasZadokStageChanged;
    }

    public void sethasZadokStageChanged(int _hasZadokStageChanged)
    {
        this.hasZadokStageChanged= _hasZadokStageChanged;
    } 
    
    public String getcurrentZadokStage()
    {
        return currentZadokStage;
    }

    public void setcurrentZadokStage(String _currentZadokStage)
    {
        this.currentZadokStage= _currentZadokStage;
    } 
    
    public int gethasFlagLeafLiguleAppeared()
    {
        return hasFlagLeafLiguleAppeared;
    }

    public void sethasFlagLeafLiguleAppeared(int _hasFlagLeafLiguleAppeared)
    {
        this.hasFlagLeafLiguleAppeared= _hasFlagLeafLiguleAppeared;
    } 
    
    public List<Double> gettilleringProfile()
    {
        return tilleringProfile;
    }

    public void settilleringProfile(List<Double> _tilleringProfile)
    {
        this.tilleringProfile= _tilleringProfile;
    } 
    
    public List<Integer> getleafTillerNumberArray()
    {
        return leafTillerNumberArray;
    }

    public void setleafTillerNumberArray(List<Integer> _leafTillerNumberArray)
    {
        this.leafTillerNumberArray= _leafTillerNumberArray;
    } 
    
    public int gettillerNumber()
    {
        return tillerNumber;
    }

    public void settillerNumber(int _tillerNumber)
    {
        this.tillerNumber= _tillerNumber;
    } 
    
    public double getcanopyShootNumber()
    {
        return canopyShootNumber;
    }

    public void setcanopyShootNumber(double _canopyShootNumber)
    {
        this.canopyShootNumber= _canopyShootNumber;
    } 
    
    public double getphyllochron()
    {
        return phyllochron;
    }

    public void setphyllochron(double _phyllochron)
    {
        this.phyllochron= _phyllochron;
    } 
    
    public double getaverageShootNumberPerPlant()
    {
        return averageShootNumberPerPlant;
    }

    public void setaverageShootNumberPerPlant(double _averageShootNumberPerPlant)
    {
        this.averageShootNumberPerPlant= _averageShootNumberPerPlant;
    } 
    
    public int getisMomentRegistredZC_39()
    {
        return isMomentRegistredZC_39;
    }

    public void setisMomentRegistredZC_39(int _isMomentRegistredZC_39)
    {
        this.isMomentRegistredZC_39= _isMomentRegistredZC_39;
    } 
    
}

File: pheno_pkg/src/cs/PhenologyState.cs
using System;
using System.Collections.Generic;
public class PhenologyState
{
    private double _leafNumber;
    private double _minFinalNumber;
    private List<DateTime> _calendarDates;
    private List<string> _calendarMoments;
    private double _ptq;
    private double _gAImean;
    private List<double> _listTTShootWindowForPTQ;
    private double _pastMaxAI;
    private List<double> _listGAITTWindowForPTQ;
    private List<double> _listPARTTWindowForPTQ;
    private List<double> _calendarCumuls;
    private double _vernaprog;
    private int _hasLastPrimordiumAppeared;
    private double _phase;
    private double _finalLeafNumber;
    private int _hasZadokStageChanged;
    private string _currentZadokStage;
    private int _hasFlagLeafLiguleAppeared;
    private List<double> _tilleringProfile;
    private List<int> _leafTillerNumberArray;
    private int _tillerNumber;
    private double _canopyShootNumber;
    private double _phyllochron;
    private double _averageShootNumberPerPlant;
    private int _isMomentRegistredZC_39;
    
    public PhenologyState()
    {
           
    }
    
    
    public PhenologyState(PhenologyState toCopy, bool copyAll) // copy constructor 
    {
    if (copyAll)
    {
    
    _leafNumber = toCopy._leafNumber;
    _minFinalNumber = toCopy._minFinalNumber;
    calendarDates = new List<DateTime>();
        for (int i = 0; i < toCopy.calendarDates.Count; i++)
        {
            calendarDates.Add(toCopy.calendarDates[i]);
        }
    
    calendarMoments = new List<string>();
        for (int i = 0; i < toCopy.calendarMoments.Count; i++)
        {
            calendarMoments.Add(toCopy.calendarMoments[i]);
        }
    
    _ptq = toCopy._ptq;
    _gAImean = toCopy._gAImean;
    listTTShootWindowForPTQ = new List<double>();
        for (int i = 0; i < toCopy.listTTShootWindowForPTQ.Count; i++)
        {
            listTTShootWindowForPTQ.Add(toCopy.listTTShootWindowForPTQ[i]);
        }
    
    _pastMaxAI = toCopy._pastMaxAI;
    listGAITTWindowForPTQ = new List<double>();
        for (int i = 0; i < toCopy.listGAITTWindowForPTQ.Count; i++)
        {
            listGAITTWindowForPTQ.Add(toCopy.listGAITTWindowForPTQ[i]);
        }
    
    listPARTTWindowForPTQ = new List<double>();
        for (int i = 0; i < toCopy.listPARTTWindowForPTQ.Count; i++)
        {
            listPARTTWindowForPTQ.Add(toCopy.listPARTTWindowForPTQ[i]);
        }
    
    calendarCumuls = new List<double>();
        for (int i = 0; i < toCopy.calendarCumuls.Count; i++)
        {
            calendarCumuls.Add(toCopy.calendarCumuls[i]);
        }
    
    _vernaprog = toCopy._vernaprog;
    _hasLastPrimordiumAppeared = toCopy._hasLastPrimordiumAppeared;
    _phase = toCopy._phase;
    _finalLeafNumber = toCopy._finalLeafNumber;
    _hasZadokStageChanged = toCopy._hasZadokStageChanged;
    _currentZadokStage = toCopy._currentZadokStage;
    _hasFlagLeafLiguleAppeared = toCopy._hasFlagLeafLiguleAppeared;
    tilleringProfile = new List<double>();
        for (int i = 0; i < toCopy.tilleringProfile.Count; i++)
        {
            tilleringProfile.Add(toCopy.tilleringProfile[i]);
        }
    
    leafTillerNumberArray = new List<int>();
        for (int i = 0; i < toCopy.leafTillerNumberArray.Count; i++)
        {
            leafTillerNumberArray.Add(toCopy.leafTillerNumberArray[i]);
        }
    
    _tillerNumber = toCopy._tillerNumber;
    _canopyShootNumber = toCopy._canopyShootNumber;
    _phyllochron = toCopy._phyllochron;
    _averageShootNumberPerPlant = toCopy._averageShootNumberPerPlant;
    _isMomentRegistredZC_39 = toCopy._isMomentRegistredZC_39;
    }
    }
    public double leafNumber
    {
        get
        {
            return this._leafNumber;
        }
        set
        {
            this._leafNumber= value;
        } 
    }
    public double minFinalNumber
    {
        get
        {
            return this._minFinalNumber;
        }
        set
        {
            this._minFinalNumber= value;
        } 
    }
    public List<DateTime> calendarDates
    {
        get
        {
            return this._calendarDates;
        }
        set
        {
            this._calendarDates= value;
        } 
    }
    public List<string> calendarMoments
    {
        get
        {
            return this._calendarMoments;
        }
        set
        {
            this._calendarMoments= value;
        } 
    }
    public double ptq
    {
        get
        {
            return this._ptq;
        }
        set
        {
            this._ptq= value;
        } 
    }
    public double gAImean
    {
        get
        {
            return this._gAImean;
        }
        set
        {
            this._gAImean= value;
        } 
    }
    public List<double> listTTShootWindowForPTQ
    {
        get
        {
            return this._listTTShootWindowForPTQ;
        }
        set
        {
            this._listTTShootWindowForPTQ= value;
        } 
    }
    public double pastMaxAI
    {
        get
        {
            return this._pastMaxAI;
        }
        set
        {
            this._pastMaxAI= value;
        } 
    }
    public List<double> listGAITTWindowForPTQ
    {
        get
        {
            return this._listGAITTWindowForPTQ;
        }
        set
        {
            this._listGAITTWindowForPTQ= value;
        } 
    }
    public List<double> listPARTTWindowForPTQ
    {
        get
        {
            return this._listPARTTWindowForPTQ;
        }
        set
        {
            this._listPARTTWindowForPTQ= value;
        } 
    }
    public List<double> calendarCumuls
    {
        get
        {
            return this._calendarCumuls;
        }
        set
        {
            this._calendarCumuls= value;
        } 
    }
    public double vernaprog
    {
        get
        {
            return this._vernaprog;
        }
        set
        {
            this._vernaprog= value;
        } 
    }
    public int hasLastPrimordiumAppeared
    {
        get
        {
            return this._hasLastPrimordiumAppeared;
        }
        set
        {
            this._hasLastPrimordiumAppeared= value;
        } 
    }
    public double phase
    {
        get
        {
            return this._phase;
        }
        set
        {
            this._phase= value;
        } 
    }
    public double finalLeafNumber
    {
        get
        {
            return this._finalLeafNumber;
        }
        set
        {
            this._finalLeafNumber= value;
        } 
    }
    public int hasZadokStageChanged
    {
        get
        {
            return this._hasZadokStageChanged;
        }
        set
        {
            this._hasZadokStageChanged= value;
        } 
    }
    public string currentZadokStage
    {
        get
        {
            return this._currentZadokStage;
        }
        set
        {
            this._currentZadokStage= value;
        } 
    }
    public int hasFlagLeafLiguleAppeared
    {
        get
        {
            return this._hasFlagLeafLiguleAppeared;
        }
        set
        {
            this._hasFlagLeafLiguleAppeared= value;
        } 
    }
    public List<double> tilleringProfile
    {
        get
        {
            return this._tilleringProfile;
        }
        set
        {
            this._tilleringProfile= value;
        } 
    }
    public List<int> leafTillerNumberArray
    {
        get
        {
            return this._leafTillerNumberArray;
        }
        set
        {
            this._leafTillerNumberArray= value;
        } 
    }
    public int tillerNumber
    {
        get
        {
            return this._tillerNumber;
        }
        set
        {
            this._tillerNumber= value;
        } 
    }
    public double canopyShootNumber
    {
        get
        {
            return this._canopyShootNumber;
        }
        set
        {
            this._canopyShootNumber= value;
        } 
    }
    public double phyllochron
    {
        get
        {
            return this._phyllochron;
        }
        set
        {
            this._phyllochron= value;
        } 
    }
    public double averageShootNumberPerPlant
    {
        get
        {
            return this._averageShootNumberPerPlant;
        }
        set
        {
            this._averageShootNumberPerPlant= value;
        } 
    }
    public int isMomentRegistredZC_39
    {
        get
        {
            return this._isMomentRegistredZC_39;
        }
        set
        {
            this._isMomentRegistredZC_39= value;
        } 
    }
}